src/main/java/com/google/devtools/build/lib/analysis/actions/PathMappers.java

```diff
@@ -134,11 +145,13 @@ public static OutputPathsMode getOutputPathsMode(
   }
 
   private static OutputPathsMode getEffectiveOutputPathsMode(
-      OutputPathsMode outputPathsMode, String mnemonic) {
-    if (outputPathsMode != OutputPathsMode.STRIP || !SUPPORTED_MNEMONICS.contains(mnemonic)) {
-      return OutputPathsMode.OFF;
+      OutputPathsMode outputPathsMode, String mnemonic, Map<String, String> executionInfo) {
+    if (outputPathsMode == OutputPathsMode.STRIP
+        && (SUPPORTED_MNEMONICS.contains(mnemonic)
+            || executionInfo.containsKey(ExecutionRequirements.SUPPORTS_PATH_MAPPING))) {
```

Comment:
@fmeum Just thinking out loud here, I wonder if we should only allow the opt-in behavior for Bazel only because within google, we probably don't want users to flip path mapping on and off as it would cause forking (mapped and unmapped paths) and potentially harm caching? @gregestren WDYT?

Reply:
Sure, I could see that make sense. I will leave that up to, happy to add this here.

Reply:
If doing so isn't too much work, can we please add the check then?

Reply:
I thought I could use `isThisBazel`, but that is only available in tests. I don't see an easy and clean way to tell apart Bazel from Blaze. I could potentially look at the execpath of the output artifact of the action, but that seems hacky. Are you aware of a better way to implement the check?

Reply:
I thought there was something similar to `JavaClassPathMode.BAZEL` but i don't think there is. I can check with others tomorrow. This shouldn't block this pr though. I will work on the import and work on the next one. Thanks!